src/snowflake/snowpark/session.py

```diff
@@ -827,12 +868,18 @@ def large_query_breakdown_enabled(self, value: bool) -> None:
         materialize the partitions, and then combine them to execute the query to improve
         overall performance.
         """
+        if threading.active_count() > 1:
```

Comment:
you can probably have a small utility function and call that for all configs with the config name as a parameter